src/indexmap.rs

```diff
@@ -0,0 +1,2539 @@
+        #[test]
+        fn extend_from_slice_doctest() {
+            let mut vec = Vec::<u8, 8>::new();
+            vec.push(1).unwrap();
+            vec.extend_from_slice(&[2, 3, 4]).unwrap();
+            assert_eq!(*vec, [1, 2, 3, 4]);
+        }
+
+        #[cfg(feature = "std")]
+        #[test]
+        fn swap_remove_doctest() {
+            let mut v: Vec<_, 8> = Vec::new();
+            v.push("foo").unwrap();
+            v.push("bar").unwrap();
+            v.push("baz").unwrap();
+            v.push("qux").unwrap();
+            assert_eq!(v.swap_remove(1), "bar");
+            assert_eq!(&*v, ["foo", "qux", "baz"]);
+            assert_eq!(v.swap_remove(0), "foo");
+            assert_eq!(&*v, ["baz", "qux"]);
+        }
+
+        #[cfg(feature = "std")]
+        #[test]
+        fn swap_remove_unchecked_doctest() {
+            let mut v: Vec<_, 8> = Vec::new();
+            v.push("foo").unwrap();
+            v.push("bar").unwrap();
+            v.push("baz").unwrap();
+            v.push("qux").unwrap();
+            assert_eq!(unsafe { v.swap_remove_unchecked(1) }, "bar");
+            assert_eq!(&*v, ["foo", "qux", "baz"]);
+            assert_eq!(unsafe { v.swap_remove_unchecked(0) }, "foo");
+            assert_eq!(&*v, ["baz", "qux"]);
+        }
+
+        #[cfg(feature = "std")]
+        #[test]
+        fn starts_with_doctest() {
+            let v: Vec<_, 8> = Vec::from_slice(b"abc").unwrap();
+            assert_eq!(v.starts_with(b""), true);
+            assert_eq!(v.starts_with(b"ab"), true);
+            assert_eq!(v.starts_with(b"bc"), false);
+        }
+
+        #[cfg(feature = "std")]
+        #[test]
+        fn ends_with_doctest() {
+            let v: Vec<_, 8> = Vec::from_slice(b"abc").unwrap();
+            assert_eq!(v.ends_with(b""), true);
+            assert_eq!(v.ends_with(b"ab"), false);
+            assert_eq!(v.ends_with(b"bc"), true);
+        }
+    }
+}
+
+// copied from https://github.com/japaric/heapless/blob/39273893deda34fa741f79438134fd4d4d7ac3ae/src/indexmap.rs
+
+use core::{borrow::Borrow, fmt, iter::FromIterator, mem, num::NonZeroU32, ops, slice};
+
+use hash32::{BuildHasher, BuildHasherDefault, FnvHasher, Hash, Hasher};
+
+use vec::Vec;
+
+/// A [`sprawl::indexmap::IndexMap`](./struct.IndexMap.html) using the default FNV hasher
+///
+/// A list of all Methods and Traits available for `FnvIndexMap` can be found in
+/// the [`sprawl::indexmap::IndexMap`](./struct.IndexMap.html) documentation.
+pub type FnvIndexMap<K, V, const N: usize> = IndexMap<K, V, BuildHasherDefault<FnvHasher>, N>;
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+struct HashValue(u16);
+
+impl HashValue {
+    fn desired_pos(&self, mask: usize) -> usize {
+        usize::from(self.0) & mask
+    }
+
+    fn probe_distance(&self, mask: usize, current: usize) -> usize {
+        current.wrapping_sub(self.desired_pos(mask) as usize) & mask
+    }
+}
+
+#[doc(hidden)]
+#[derive(Clone)]
+pub struct Bucket<K, V> {
```

Comment:
These should all be pub(crate).

Reply:
I would only make `IndexMap` pub here.

Reply:
I made only the things that needed to be pub(crate) or pub those things and the rest I made either pub(super) or removed pub. The sprawl tests pass, and the bevy ui and button examples work. Let me know if my public-ness is correct.